### DISCUSSION TITLE
Issue accessing RealSense Device

### DISCUSSION THREAD

**Discussion by BettaDayz** [bonsai-rx/bonsai#2087] [Q&A] [answered]
Hi all!

Im currently trying to use bonsai with the RealSense package, using the most simple workflow (i.e. RealSense Device as source, then accessing color and depth stream), but I get an error when I try to start the workflow.  I have copy-pasted the code below:

Bonsai.WorkflowRuntimeException: Error calling 'rs_get_device' with context:0000000021CD9290, index:0. out of range value for argument "index" ---> RealSense.Net.RealSenseException: Error calling 'rs_get_device' with context:0000000021CD9290, index:0. out of range value for argument "index"
   at RealSense.Net.Native.NativeHelper.ThrowExceptionForRsError(IntPtr error)
   at RealSense.Net.DeviceCollection.get_Item(Int32 index)
   at Bonsai.RealSense.RealSenseDevice.<Generate>b__0()
   at System.Reactive.Linq.ObservableImpl.Defer`1._.Run()
   --- End of inner exception stack trace ---

Is there anything I can do to mitigate this error?  I am using a D435 RealSense camera.

**Comment by banchan86** [comment, marked as answer]
Hi @BettaDayz can you try the pre-release version 0.3 of the package? Uninstall the released version 0.2.0 first, and then using the package manager, check the box that says `Include prerelease`, and search for `RealSense` with the package source - `Bonsai Packages`. It should bring up a package with the official red bonsai logo and version 0.3.0-build3440 (Prerelease). If it works, can you give an update here? It seems the last update on this topic was from https://groups.google.com/g/bonsai-users/c/IGVoEvjqjUw/m/lzF6mDHZAgAJ and I'm not sure if the prerelease version fixes the issue.

**Reply by BettaDayz** [reply]
Hi @banchan86 !

Thanks! I dont get the index error anymore!
Hope all is well, happy new year!
